src/main/java/elements/CheckBoxImpl.java

```diff
@@ -0,0 +1,44 @@
+package elements;
+
+
+import org.openqa.selenium.ElementNotVisibleException;
+import org.openqa.selenium.WebElement;
+import util.ScrollUtils;
+import util.WaitUtils;
+
+
+/***
+ * Wrapper class with Checkbox functionalities
+ */
+public class CheckBoxImpl extends ElementImpl implements Checkbox {
+
+    public CheckBoxImpl(WebElement element) {
+        super(element);
+    }
+
+    @Override
+    public void clickCheckbox() {
+        try {
+            changeState();
+        } catch (ElementNotVisibleException notVisibleException) {
+            try {
+                WaitUtils.waitForVisibility(getWrappedElement());
+                changeState();
+            } catch (Exception ex) {
+                ScrollUtils.scrollToElementView(getWrappedElement());
+                WaitUtils.waitForVisibility(getWrappedElement());
+                changeState();
+            }
+        }
+    }
+
+    @Override
+    public boolean isSelected() {
+        return this.getWrappedElement().isSelected();
+    }
+
+    @Override
+    public void changeState() {
```

Comment:
In my opinion it would be better to create method `changeState (boolean expectedState)` which will set checkbox value according to the state we want it to be, what do you think?

Default checkbox states might change in the app and it will lead to test failures
Plus current method implementation doesn't give us information about the checkbox state, the only thing we know about it is that it's changed